rpc/core/events.go

```diff
@@ -100,18 +95,31 @@ func Subscribe(wsCtx rpctypes.WSRPCContext, query string) (*ctypes.ResultSubscri
 func Unsubscribe(wsCtx rpctypes.WSRPCContext, query string) (*ctypes.ResultUnsubscribe, error) {
 	addr := wsCtx.GetRemoteAddr()
 	logger.Info("Unsubscribe from query", "remote", addr, "query", query)
-	q, ok := wsCtx.DeleteSubscription(query)
-	if !ok {
-		return nil, errors.New("subscription not found")
+	q, err := tmquery.New(query)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse query")
+	}
+	err = eventBusFor(wsCtx).Unsubscribe(context.Background(), addr, q)
+	if err != nil {
+		return nil, err
 	}
-	eventBus.Unsubscribe(context.Background(), addr, q.(*tmquery.Query))
 	return &ctypes.ResultUnsubscribe{}, nil
 }
 
 func UnsubscribeAll(wsCtx rpctypes.WSRPCContext) (*ctypes.ResultUnsubscribe, error) {
 	addr := wsCtx.GetRemoteAddr()
 	logger.Info("Unsubscribe from all", "remote", addr)
-	eventBus.UnsubscribeAll(context.Background(), addr)
-	wsCtx.DeleteAllSubscriptions()
+	err := eventBusFor(wsCtx).UnsubscribeAll(context.Background(), addr)
+	if err != nil {
+		return nil, err
+	}
 	return &ctypes.ResultUnsubscribe{}, nil
 }
+
+func eventBusFor(wsCtx rpctypes.WSRPCContext) tmtypes.EventBusSubscriber {
+	es := wsCtx.GetEventSubscriber()
+	if es == nil {
+		es = eventBus
```

Comment:
:o hmm.

So I guess this happens if someone calls NewWebsocketManager and forgets to pass the option to set the eventBus? 

Also maybe its time to make all these funcs methods on a `RPCPipe` struct or something so we can do away with the globals. Agree ?